internal/repository/database/init.go

```diff
@@ -0,0 +1,57 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/eurofurence/reg-auth-service/internal/repository/config"
+	"github.com/eurofurence/reg-auth-service/internal/repository/database/dbrepo"
+	"github.com/eurofurence/reg-auth-service/internal/repository/database/inmemorydb"
+	"github.com/eurofurence/reg-auth-service/internal/repository/logging"
+)
+
+var (
+	ActiveRepository dbrepo.Repository
+	pruneTicker      *time.Ticker
+	pruneStop        chan bool
+)
+
+// only exported so you can use it in test code - use Open()
+func SetRepository(repository dbrepo.Repository) {
+	ActiveRepository = repository
+}
+
+func Open() {
+	var r dbrepo.Repository
+	logging.NoCtx().Info("Opening inmemory database...")
+	r = inmemorydb.Create()
```

Comment:
I'm surprised this operation can't fail. Would have expected Out-of-Memory or similar.

Reply:
While it theoretically could, it does not pre-allocate any storage when `inmemorydb.Create()` is called, so this call itself is highly unlikely to fail (which does ofc not mean that adding entries to it can't fail at a later point due to oom).